frontend/ui/opdsparser.lua

```diff
@@ -84,7 +84,7 @@ function OPDSParser:parse(text)
         return s:gsub( "%p", {["&"] = "&amp;", ["<"] = "&lt;", [">"] = "&gt;" } )
     end )
     local xlex = luxl.new(text, #text)
-    return self:createFlatXTable(xlex)
+    return assert(self:createFlatXTable(xlex))
```

Comment:
Canned response, but is this assert protected by a pcall? ;-)

Reply:
Do you suggest to use 
`assert(pcall(createFleXTable, xlex))` ?

Reply:
I suggest whatever doesn't crash when the assertion fails, which would be the other way around (`pcall(assert(`). ;-) Note that we also have a debug.assert function (might be named slightly differently) which will crash the program in debug mode.

I'm not sufficiently familiar with the code to know what makes the most sense here specifically. It could also be a level higher up, like: `pcall(OPDSParser.parse, text)`.

Reply:
> I suggest whatever doesn't crash when the assertion fails, which would be the other way around (pcall(assert(). ;-) Note that we also have a debug.assert function (might be named slightly differently) which will crash the program in debug mode.


Example please. I really don't understand :)

>I'm not sufficiently familiar with the code to know what makes the most sense here specifically. It could also be a level higher up, like: pcall(OPDSParser.parse, text).


pcall(OPDSParser.parse, text) won't work. It can't enter to any opds server.

Reply:
I mean `dbg.dassert()`. But that's not really appropriate here.

https://github.com/koreader/koreader/blob/0885e9947dead0fe4b30d816ffeafca4a3932a19/frontend/dbg.lua#L50-L53

> pcall(OPDSParser.parse, text) won't work. It can't enter to any opds server.

How do you mean? Something like `pcall(OPDSParser.parse, OPDSParser, text)` or `pcall(OPDSParser.parse, self, text)`, whatever works in the relevant context.

My point is simply that `assert()` means the program halts unless it's taken care of in some way.

```
> function bla() assert(false) end
> bla()
stdin:1: assertion failed!
stack traceback:
	[C]: in function 'assert'
	stdin:1: in function 'bla'
	stdin:1: in main chunk
	[C]: in ?
```

Reply:
I was going to ask : but why add that `assert()` ?! Why would youd need that?
But following the code, the only place this is called is actually wrapped in a pcall():
https://github.com/koreader/koreader/blob/0885e9947dead0fe4b30d816ffeafca4a3932a19/frontend/ui/widget/opdsbrowser.lua#L324-L363
So, you actually added it to make that pcall() fail.
So, case closed :)

Reply:
Excellent, so the answer to my initial question is a straightforward "yes". :+1: